pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -924,38 +1000,52 @@ func (r *ReconcileInfinispan) createSiteService(siteServiceName string, infinisp
 }
 
 func appendBackupSite(xsite *ispnutil.XSite, infinispan *infinispanv1.Infinispan, backup *infinispanv1.InfinispanSitesBackupSpec, logger logr.Logger) error {
-	backupSiteURL, err := url.Parse(backup.URL)
+	restConfig, err := GetRemoteSiteRESTConfig(infinispan, backup, logger)
 	if err != nil {
 		return err
 	}
-	switch scheme := backupSiteURL.Scheme; scheme {
-	case "minikube":
-		logger.Info("connect to backup minikube cluster", "url", backupSiteURL)
-		// Copy URL to make modify it for backup access
-		copyURL, _ := url.Parse(backupSiteURL.String())
-		copyURL.Scheme = "https"
-		restConfig, err := GetVanillaKubernetesRESTConfig(copyURL.String(), backup.Secret, infinispan, logger)
-		if err != nil {
-			return err
-		}
 
-		remoteKubernetes, err := ispnutil.NewKubernetesFromConfig(restConfig)
-		if err != nil {
-			logger.Error(err, "could not connect to master URL", "URL", copyURL.String())
-			return err
-		}
+	remoteKubernetes, err := ispnutil.NewKubernetesFromConfig(restConfig)
+	if err != nil {
+		logger.Error(err, "could not connect to master URL", "URL", backup.URL)
+		return err
+	}
 
-		err = appendKubernetesBackupSite(xsite, infinispan, backup, remoteKubernetes, logger)
-		if err != nil {
-			return err
-		}
-		return nil
+	err = appendKubernetesBackupSite(xsite, infinispan, backup, remoteKubernetes, logger)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetRemoteSiteRESTConfig(infinispan *infinispanv1.Infinispan, backup *infinispanv1.InfinispanSitesBackupSpec, logger logr.Logger) (*restclient.Config, error) {
```

Comment:
Does this need to be exportable? Also GetMinikubeRESTConfig and GetOpenShiftRESTConfig

Reply:
Doesn't need to, I'll fix it.